suite-common/wallet-core/src/fiat-rates/fiatRatesSelectors.ts

```diff
@@ -121,12 +136,24 @@ export const selectTransactionsWithMissingRates = (
     localCurrency: FiatCurrencyCode,
 ) => {
     const accountTransactions = selectTransactions(state);
+    const historicFiatRates = selectHistoricFiatRates(state);
 
     return pipe(
         accountTransactions,
         D.mapWithKey((accountKey, txs) => ({
             account: selectAccountByKey(state, accountKey as AccountKey),
-            txs: txs.filter(tx => !tx.rates?.[localCurrency]),
+            txs: txs.filter(tx => {
+                const fiatRateKey = getFiatRateKey(
+                    tx.symbol,
+                    localCurrency as FiatCurrencyCode,
+                    tx.tokens[0]?.contract as TokenAddress,
```

Comment:
What if there is more than one token in transaction?

Reply:
It should not be possible to have more tokens in one tx

Reply:
Fyi: We might not parse that now from txns, but on Solana you can:

https://solscan.io/tx/3dVSJKxrBVVRwnndYpxuWS16Nnj3xiMrLbWqhu2XP5KpBAZri4nA2juFVyFoEWtVzuhD4DgcKgCADzXpXcBJPgJp

Reply:
You can have unlimited number of tokens in one tx.

Reply:
Ok but we have quite some places where we use this approach, it will be good to check if it isn't causing any bug https://github.com/trezor/trezor-suite/issues/12384